### PR TITLE
[13.0][ADD] website_sale_coupon_page: New module to show the promotions on website

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT30853

please @joao-p-marques @victoralmau review it

Set as WIP because I will add some test

### CODE REVIEW

File: website_sale_coupon_page/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                    ("is_published", "=", True),
+                    "|",
+                    ("website_id", "=", False),
+                    ("website_id", "=", request.env.context.get("website_id")),
+                ]
+            )
+        )
+        promos = request.env["sale.coupon.program"]
+        for promo in all_promos:
+            if promo._is_valid_partner(active_partner_id):
```
Comment (victoralmau) [inline website_sale_coupon_page/controllers/main.py]:
```suggestion
        all_promos = (
            request.env["sale.coupon.program"]
            .with_user(request.env.user)
            .search(
                [
                    ("is_published", "=", True),
                    "|",
                    ("website_id", "=", False),
                    ("website_id", "=", request.env["website"].get_current_website()),
                ]
            )
        )
        promos = request.env["sale.coupon.program"]
        for promo in all_promos:
            if promo._is_valid_partner(request.env.user.partner_id):
```
I don't really sure about all suggestion but probably simplify and improve code if some user logged show only allowed (if some addon filter promos).

**Review by victoralmau — CHANGES_REQUESTED** [review]
Is possible to add some portal tests?

**Comment by CarlosRoca13** [comment]
@victoralmau changes done and test added

File: website_sale_coupon_page/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteSale(http.Controller):
+    @http.route(["""/promotions"""], type="http", auth="public", website=True)
+    def promotion(self, **post):
+        all_promos = (
+            request.env["sale.coupon.program"]
+            .sudo()
```
Comment (victoralmau) [inline website_sale_coupon_page/controllers/main.py:12]:
```suggestion
            .with_user(request.env.user)
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: website_sale_coupon_page/controllers/main.py (hunk 2 above)
Comment (chienandalu) [reply website_sale_coupon_page/controllers/main.py:12]: Public or portal users don't have access to this model

**Review by victoralmau — APPROVED** [review]
Code review OK.

**Comment by pedrobaeza** [comment]
This should go to the new repository to be created OCA/sale-promotion

**Comment by pedrobaeza** [comment]
You can move it now: https://github.com/OCA/sale-promotion

**Pull request closed without merging**